tests/test.sh

```diff
@@ -613,4 +613,13 @@ fi
 
 rm -rf "$TEST_TEMP_DIR"
 
+# now we need to strip the date header as it is undeterministic
+find -type f -name "*.headers" -exec sed --regexp-extended --in-place "s/date: .+//" {} \;
```

Comment:
ah, silly me, this step here is why you only have the stable things. My only minor gripe is that we have to use `find` here. Instead, lets create a dedicated `curl_with_headers $URL $FILE` function that downloads URL into a FILE, and also saves FILE.headers and removes unneeded stuff from it.  This way if there is an error, the header files will be in the right state right away, and whoever is debugging it won't get confused why expected and resulting files are different

Reply:
P.S. For this line, you can do fixes after the fact, i.e. instead of this:

```
$CURL  --dump-header  "$FILENAME.headers" "$URL" | jq --sort-keys -e 'walk(if type == "number" then .+0.0 else . end)' > "$FILENAME"
```

do this:

```
curl_with_headers "$URL" "$FILENAME"
cat "$FILENAME" jq --sort-keys -e 'walk(if type == "number" then .+0.0 else . end)' > "${FILENAME}__tmp"
mv "$FILENAME__tmp" "$FILENAME"
```

Reply:
I think `curl_with_headers` is a bit too non-obvious in what it is doing.

I changed it to the following instead.

```sh
$CURL --dump-header  "$FILENAME.headers" "$URL" > "$FILENAME"
clean_headers_dump "$FILENAME.headers"
```